drivers/serial/uart_nrf_sw_lpuart.c

```diff
@@ -296,14 +349,19 @@ static void activate_rx(struct lpuart_data *data)
 		return;
 	}
 
+	LOG_DBG("RX: Ready");
+	data->rx_got_data = false;
+	data->rx_state = RX_ACTIVE;
 	err = uart_rx_enable(data->uart, data->rx_buf,
 				data->rx_len, data->rx_timeout);
 	__ASSERT(err == 0, "RX: Enabling failed (err:%d)", err);
 
 	/* Ready. Confirm by toggling the pin. */
-	rdy_pin_blink(data);
-	LOG_DBG("RX: Ready");
-	data->rx_state = RX_ACTIVE;
+	if (rdy_pin_blink(data) == false) {
```

Comment:
Nitpicking (might be corrected for consistency):
```suggestion
	if (!rdy_pin_blink(data)) {
```
also https://github.com/nrfconnect/sdk-nrf/blob/0b8bbed8cf662cc7eccbdf6d1baf0a5846dacce1/drivers/serial/uart_nrf_sw_lpuart.c#L549 https://github.com/nrfconnect/sdk-nrf/blob/0b8bbed8cf662cc7eccbdf6d1baf0a5846dacce1/drivers/serial/uart_nrf_sw_lpuart.c#L696

Reply:
done